### PR TITLE
DeviceInfo: Add device information to header and format device details

### PR DESCRIPTION
**Pull request by reyhanlioglu**
Also add makefile

## Summary by CodeRabbit

- **New Features**
	- Introduced comprehensive build and development commands in the Makefile for a Flutter project.
	- Enhanced device information handling, including model, version, and platform details in network headers.
	- Added device version field in device registration requests.

- **Refactor**
	- Updated network manager to decode and include detailed device information.
	- Modified device utility classes to provide consistent device information across platforms.
	- Refactored device registration to utilize comprehensive device information objects.

- **Documentation**
	- Updated export paths and import statements to reflect changes in device utility access and structure.

### CODE REVIEW

File: lib/core/network/models/neo_network_header_key.dart [hunk 1]
```diff
@@ -11,17 +11,18 @@
  */
 
 abstract class NeoNetworkHeaderKey {
-  static const contentType = "Content-Type";
   static const acceptLanguage = "Accept-Language";
-  static const contentLanguage = "Content-Language";
+  static const accessToken = "access_token";
```
Comment (bgoktugozdemir-dev) [inline lib/core/network/models/neo_network_header_key.dart:15]: Bu zaten Authorization'da gitmiyor mu?

File: lib/core/util/device_util/device_util.dart [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        platform: getPlatformName(),
+        version: androidInfo.version.release,
+      );
+    } else {
+      return null;
+    }
+  }
+
+  String getPlatformName() {
+    if (kIsWeb) {
```
Comment (bgoktugozdemir-dev) [inline lib/core/util/device_util/device_util.dart:58]: Bunun yerine direkt aşağıdakini return etsek olmaz mı?
`Platform.operatingSystem`

Comment (reyhanlioglu) [reply lib/core/util/device_util/device_util.dart:58]: bu method "ios" gibi dönüyor bana "iOS" yazması lazım, android için de aynı şekilde büyük küçük harf önemli

File: lib/core/network/models/neo_network_header_key.dart (hunk 1 above)
Comment (reyhanlioglu) [reply lib/core/network/models/neo_network_header_key.dart:15]: GetWorkflowQueryParametersUseCase methodunda özellikle bu keyle isteniyor yeni eklenmiş bir şey değil

**Review by bgoktugozdemir-dev — APPROVED** [review]

**Pull request merged**